[BUG] :bug: Creating a group on a non existing application should return 404 instead of 500

When a user tries to create a group on a non existing application (ex : POST on /realms/domaine1/applications/donotexist/groups) sugoi throws a RuntimeException :

```
java.lang.RuntimeException: Failed to create group test_donotexist
at [fr.insee.sugoi.store](http://fr.insee.sugoi.store/).ldap.LdapWriterStore.createGroup([LdapWriterStore.java:232](http://ldapwriterstore.java:232/))
at fr.insee.sugoi.core.service.impl.GroupServiceImpl.create([GroupServiceImpl.java:51](http://groupserviceimpl.java:51/))
at [fr.insee.sugoi.services](http://fr.insee.sugoi.services/).controller.GroupController.createGroups([GroupController.java:179](http://groupcontroller.java:179/))
at [fr.insee.sugoi.services](http://fr.insee.sugoi.services/).controller.GroupController$$FastClassBySpringCGLIB$$37c88f08.invoke(<generated>)
....
Caused by: com.unboundid.ldap.sdk.LDAPException: no such object
at com.unboundid.ldap.sdk.LDAPConnection.add([LDAPConnection.java:2221](http://ldapconnection.java:2221/))
at com.unboundid.ldap.sdk.AbstractConnectionPool.add([AbstractConnectionPool.java:798](http://abstractconnectionpool.java:798/))
at [fr.insee.sugoi.store](http://fr.insee.sugoi.store/).ldap.LdapWriterStore.createGroup([LdapWriterStore.java:222](http://ldapwriterstore.java:222/))
... 119 common frames omitted
```
and a 500 is returned to the user.
Instead of this, a check should be done before trying to create the group and if the application does not exist an ApplicationNotFoundException should be thrown thus returning a 404 to the user.

This issue should come with more tests to avoid future regression. If behaviour expected are not obvious, it should be written in the javadoc

--- a/sugoi-api-file-store-provider/src/test/resources/sugoi-file-tests/apps/Applitest
+++ b/sugoi-api-file-store-provider/src/test/resources/sugoi-file-tests/apps/Applitest
@@ -37,9 +37,9 @@
                "habilitations":[
                   
                ],
-               "address":[
+               "address":{
 
-               ],
+               },
                "metadatas":{
                   
                },

--- a/sugoi-api-file-store-provider/src/test/resources/sugoi-file-tests/orgs/testo
+++ b/sugoi-api-file-store-provider/src/test/resources/sugoi-file-tests/orgs/testo
@@ -1,1 +1,29 @@
- {"identifiant":"testo","gpgkey":null,"organization":{"identifiant":"testi","gpgkey":null,"organization":null,"address":{"line1":"Insee","line4":"88 AVE VERDIER"},"metadatas":{},"attributes":{}},"address":["Insee","","","88 AVE VERDIER"],"metadatas":{},"attributes":{"description":"Insee"}} 
+{
+   "identifiant":"testo",
+   "gpgkey":null,
+   "organization":{
+      "identifiant":"testi",
+      "gpgkey":null,
+      "organization":null,
+      "address":{
+         "line1":"Insee",
+         "line4":"88 AVE VERDIER"
+      },
+      "metadatas":{
+
+      },
+      "attributes":{
+
+      }
+   },
+   "address":{
+      "line1":"Insee",
+      "line4":"88 AVE VERDIER"
+   },
+   "metadatas":{
+
+   },
+   "attributes":{
+      "description":"Insee"
+   }
+}

--- a/sugoi-api-ldap-store-provider/src/main/java/fr/insee/sugoi/store/ldap/LdapWriterStore.java
+++ b/sugoi-api-ldap-store-provider/src/main/java/fr/insee/sugoi/store/ldap/LdapWriterStore.java
@@ -210,7 +210,12 @@ public class LdapWriterStore extends LdapStore implements WriterStore {
   @Override
   public ProviderResponse createGroup(
       String appName, Group group, ProviderRequest providerRequest) {
+
     try {
+      if (ldapReaderStore.getApplication(appName).isEmpty()) {
+        throw new ApplicationNotFoundException(appName);
+      }
+
       if (ldapReaderStore.getGroup(appName, group.getName()).isPresent()) {
         throw new GroupAlreadyExistException(
             String.format(

--- a/sugoi-api-ldap-store-provider/src/test/java/fr/insee/sugoi/ldap/LdapWriterStoreTest.java
+++ b/sugoi-api-ldap-store-provider/src/test/java/fr/insee/sugoi/ldap/LdapWriterStoreTest.java
@@ -23,6 +23,7 @@ import fr.insee.sugoi.core.model.ProviderResponse;
 import fr.insee.sugoi.core.model.ProviderResponse.ProviderResponseStatus;
 import fr.insee.sugoi.ldap.utils.config.LdapConfigKeys;
 import fr.insee.sugoi.model.*;
+import fr.insee.sugoi.model.exceptions.ApplicationNotFoundException;
 import fr.insee.sugoi.model.exceptions.InvalidPasswordException;
 import fr.insee.sugoi.model.exceptions.StoragePolicyNotMetException;
 import fr.insee.sugoi.model.fixtures.StoreMappingFixture;
@@ -375,6 +376,17 @@ public class LdapWriterStoreTest {
         "Should retrieve Groupy",
         ldapReaderStore.getGroup("Applitest", "Groupy_Applitest").get().getName(),
         is("Groupy_Applitest"));
+  }
+
+  @Test
+  void testCreateGroupApplicationNotFound() {
+    Group group = new Group();
+    assertThrows(
+        ApplicationNotFoundException.class,
+        () -> {
+          ldapWriterStore.createGroup("Donotexist", group, null);
+        },
+        "Should throw ApplicationNotFoundException");
   }
 
   @Test

